src/core/p5.Renderer2D.js

```diff
@@ -804,14 +745,7 @@ class Renderer2D extends p5.Renderer {
         bl = hh;
       }
 
-      // Draw shape
-      if (!this._clipping) ctx.beginPath();
-      ctx.moveTo(x + tl, y);
-      ctx.arcTo(x + w, y, x + w, y + h, tr);
-      ctx.arcTo(x + w, y + h, x, y + h, br);
-      ctx.arcTo(x, y + h, x, y, bl);
-      ctx.arcTo(x, y, x + w, y, tl);
-      ctx.closePath();
+      ctx.roundRect(x, y, w, h, [tl, tr, br, bl]);
```

Comment:
On bad usage (negative radii) this produces a different error than arcTo, so it is not caught correctly. I was not able to find where to make a corresponding update. I looked in the `friendly_errors` directory, particularly in `stacktrace.js`, but was not sure how to proceed.